source/uwp/Renderer/lib/ActionHelpers.cpp

```diff
@@ -43,6 +43,43 @@ namespace AdaptiveNamespace::ActionHelpers
         return S_OK;
     }
 
+    void SetTooltip(HSTRING toolTipText, IDependencyObject* tooltipTarget)
+    {
+        if (toolTipText != nullptr)
+        {
+            ComPtr<ITextBlock> tooltipTextBlock =
+                XamlHelpers::CreateXamlClass<ITextBlock>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_TextBlock));
+            THROW_IF_FAILED(tooltipTextBlock->put_Text(toolTipText));
+
+            ComPtr<IToolTip> toolTip =
+                XamlHelpers::CreateXamlClass<IToolTip>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_ToolTip));
+            ComPtr<IContentControl> toolTipAsContentControl;
+            THROW_IF_FAILED(toolTip.As(&toolTipAsContentControl));
+            THROW_IF_FAILED(toolTipAsContentControl->put_Content(tooltipTextBlock.Get()));
+
+            ComPtr<IToolTipServiceStatics> toolTipService;
+            THROW_IF_FAILED(GetActivationFactory(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_ToolTipService).Get(),
+                                                 &toolTipService));
+
+            THROW_IF_FAILED(toolTipService->SetToolTip(tooltipTarget, toolTip.Get()));
+        }
+    }
+
+    void SetAutomationNameAndDescription(IDependencyObject* dependencyObject, HSTRING name, HSTRING description)
```

Comment:
Should we verify that `dependencyObject != nullptr` here too since I'm assuming that `SetName` and the like will dereference it?

Reply:
So, the tooltip check in the above was because if there isn't a tooltip set in the card we don't want to create an empty tooltip in the UI. In general, we're not doing null parameter checking on required parameters in internal methods in this code. 

One thing that would have helped make this clearer though is that i'm missing SAL annotations on the parameters. I'll add those so it's more obvious what's going on.